src/khoj/processor/tools/online_search.py

```diff
@@ -51,4 +70,41 @@ def _search_with_google(subquery: str):
         logger.info(f"Searching with Google for '{subquery}'")
         response_dict[subquery] = _search_with_google(subquery)
 
+    extracted_content: Dict[str, List] = {}
+    if is_none_or_empty(OLOSTEP_API_KEY):
+        logger.warning("OLOSTEP_API_KEY is not set. Skipping web scraping.")
+        return response_dict
+
+    for subquery in response_dict:
+        # If a high quality answer is not found, search the web pages of the first 3 organic results
+        if is_none_or_empty(response_dict[subquery].get("answerBox")):
+            extracted_content[subquery] = []
+            for result in response_dict[subquery].get("organic")[:1]:
+                logger.info(f"Searching web page of '{result['link']}'")
+                try:
+                    extracted_content[subquery].append(search_with_olostep(result["link"]).strip())
+                except Exception as e:
+                    logger.error(e, exc_info=True)
```

Comment:
Maybe good to prefix a description of the error for context
```suggestion
                    logger.error(f'Failed to use Olostep to search through {result["link"]}.\nException: {e}", exc_info=True)
```